### PR TITLE
compose serializers

### PR DESCRIPTION
**Pull request by craigbilner**
This is the start of being able to curate snapshots in conjunction with `jest-lint`. `jest-serializer` has now been split into separate functionality and applied to each package at a minimum in order to preserve the current snapshots.

The next steps would be to iterate on `jest-lint` and the various `serializers` in order to make "good" snapshots as easy as possible. The first obvious feature being, blitzing external components to that involve integration headaches

The PR is rather large because it adds serializers to every package but you can focus on any changes to existing snapshots which is the important part. For example `ad` now has an example where `rnw` styles are printed in the snapshot where the current snapshots are fairly meaningless

For the more interested `jest-serializer` has the actual functionality change where we have a traverser, printers and transformer to allow us to curate the snapshots on a case by case basis

breaking change: add options object to jest-configurator
feat: hoist dev specific rnw styles in snapshots a la styled components
feat: remove empty values from snapshots e.g. stickyHeaderIndices
feat: web retains className even if empty and native does not
chore: remove coverage flags from test scripts
chore: bump enzyme-to-json

### CODE REVIEW

File: packages/article-label/package.json [hunk 1]
```diff
@@ -53,7 +54,7 @@
     "babel-plugin-styled-components": "1.5.1",
     "babel-preset-react-native": "4.0.0",
     "enzyme": "3.3.0",
-    "enzyme-to-json": "3.3.0",
+    "enzyme-to-json": "3.3.4",
```
Comment (nikhedonia) [inline packages/article-label/package.json]: to we use `enzyme-to-json` directly?

File: packages/article-list/__tests__/android/serializers.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const {
+  compose,
+  minimalNativeTransform,
+  flattenStyleTransform,
+  print
+} = require("@times-components/jest-serializer");
+const { createSerializer } = require("enzyme-to-json");
+
+expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
```
Comment (nikhedonia) [inline packages/article-list/__tests__/android/serializers.js]: I expected this to be inside compose. Maybe it makes sense to export this serializer from the `jest-serialize` package?

**Comment by nikhedonia** [comment]
Jest-serializer would benefit from a readme that describes the various serializes and how they compose.

File: packages/ad/__tests__/web/__snapshots__/placeholder.test.js.snap [hunk 3]
```diff
@@ -1,8 +1,19 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Placeholder renders a large placeholder 1`] = `
+<style>
+{
+  "S1": {
```
Comment (nikhedonia) [inline packages/ad/__tests__/web/__snapshots__/placeholder.test.js.snap]: would be prettier if it actually would look like css.

File: packages/card/__tests__/web/__snapshots__/card.web.test.js.snap [hunk 4]
```diff
@@ -14,6 +14,7 @@ exports[`1. Renders default layout 1`] = `
     }
   >
     <View
+      className=""
```
Comment (nikhedonia) [inline packages/card/__tests__/web/__snapshots__/card.web.test.js.snap]: maybe get rid of those empty classNames

File: packages/jest-serializer/__tests__/web/__snapshots__/flatten-style.test.js.snap [hunk 5]
```diff
@@ -0,0 +1,16 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`The flatten style serializer should for web flatten styles 1`] = `
```
Comment (nikhedonia) [inline packages/jest-serializer/__tests__/web/__snapshots__/flatten-style.test.js.snap]: maybe "The flatten style serializer for web should flatten styles 1" ?

File: packages/ad/__tests__/web/__snapshots__/placeholder.test.js.snap (hunk 3 above)
Comment (craigbilner) [reply packages/ad/__tests__/web/__snapshots__/placeholder.test.js.snap]: Have raised #1031

File: packages/card/__tests__/web/__snapshots__/card.web.test.js.snap (hunk 4 above)
Comment (craigbilner) [reply packages/card/__tests__/web/__snapshots__/card.web.test.js.snap]: will remove in the future, this "proves" that the snapshots aren't correct

File: packages/jest-serializer/__tests__/web/__snapshots__/flatten-style.test.js.snap (hunk 5 above)
Comment (craigbilner) [reply packages/jest-serializer/__tests__/web/__snapshots__/flatten-style.test.js.snap]: will pick up

**Pull request merged**